### PR TITLE
Add Support to Delete Resources

### PR DESCRIPTION
**Pull request by ImalshaG**
This PR adds support to delete resources and refactors and adds code improvements.

Related Issue: https://github.com/wso2/product-is/issues/15461

### CODE REVIEW

File: iamctl/pkg/userStores/export.go [hunk 1]
```diff
@@ -75,66 +79,38 @@ func exportUserStore(userStoreId string, outputDirPath string, format string) er
...
-		if err != nil {
-			return fmt.Errorf("error while parsing the content disposition header: %s", err)
-		}
 
-		fileName := params["filename"]
-		exportedFileName := filepath.Join(outputDirPath, fileName)
-		fileInfo := utils.GetFileInfo(exportedFileName)
+	fileName := params["filename"]
+	exportedFileName := filepath.Join(outputDirPath, fileName)
+	fileInfo := utils.GetFileInfo(exportedFileName)
```
Comment (chamathns) [inline iamctl/pkg/userStores/export.go:95]: are the possible exceptions here handled through the utils method?

File: iamctl/pkg/userStores/import.go [hunk 2]
```diff
@@ -88,3 +95,21 @@ func importUserStore(userStoreId string, importFilePath string) error {
 	log.Println("User store imported successfully.")
 	return nil
 }
+
+func removeDeletedDeployedUserstores(localFiles []os.FileInfo, deployedUserstores []userStore) {
+
+	// Remove deployed user stores that do not exist locally.
+deployedResourcess:
```
Comment (chamathns) [inline iamctl/pkg/userStores/import.go:102]: check formatting

File: iamctl/pkg/utils/commonUtils.go [hunk 3]
```diff
@@ -59,6 +61,55 @@ func getResourcePath(resourceType string) string {
...
 }
 
+func SendExportRequest(resourceId, fileType, resourceType string, excludeSecrets bool) (resp *http.Response, err error) {
+
+	var reqUrl string
+	if resourceType == APPLICATIONS {
+		reqUrl = SERVER_CONFIGS.ServerUrl + "/t/" + SERVER_CONFIGS.TenantDomain + "/api/server/v1/" + getResourcePath(resourceType) + "/" + resourceId + "/exportFile"
+	} else {
+		reqUrl = SERVER_CONFIGS.ServerUrl + "/t/" + SERVER_CONFIGS.TenantDomain + "/api/server/v1/" + getResourcePath(resourceType) + "/" + resourceId + "/export"
+	}
```
Comment (chamathns) [inline iamctl/pkg/utils/commonUtils.go]: let's extract this out to a separate method. something like buildRequestUrl

File: iamctl/pkg/utils/commonUtils.go [hunk 4]
```diff
@@ -125,7 +176,12 @@ func SendImportRequest(importFilePath, fileData, resourceType string) error {
 
 func SendUpdateRequest(resourceId, importFilePath, fileData, resourceType string) error {
 
-	reqUrl := SERVER_CONFIGS.ServerUrl + "/t/" + SERVER_CONFIGS.TenantDomain + "/api/server/v1/" + getResourcePath(resourceType) + "/" + resourceId + "/import"
+	var reqUrl string
+	if resourceType == APPLICATIONS {
+		reqUrl = SERVER_CONFIGS.ServerUrl + "/t/" + SERVER_CONFIGS.TenantDomain + "/api/server/v1/" + getResourcePath(resourceType) + "/import"
+	} else {
+		reqUrl = SERVER_CONFIGS.ServerUrl + "/t/" + SERVER_CONFIGS.TenantDomain + "/api/server/v1/" + getResourcePath(resourceType) + "/" + resourceId + "/import"
+	}
```
Comment (chamathns) [inline iamctl/pkg/utils/commonUtils.go]: we could think of reusing the buildRequestUrl method here as well

File: iamctl/pkg/utils/commonUtils.go [hunk 5]
```diff
@@ -187,3 +243,46 @@ func SendUpdateRequest(resourceId, importFilePath, fileData, resourceType string
 	}
 	return fmt.Errorf("unexpected error when importing resource: %s", resp.Status)
 }
+
+func Contains(slice []string, item string) bool {
+	for _, s := range slice {
```
Comment (chamathns) [inline iamctl/pkg/utils/commonUtils.go:43]: add blank line

File: iamctl/pkg/userStores/export.go (hunk 1 above)
Comment (ImalshaG) [reply iamctl/pkg/userStores/export.go:95]: This method simply extracts the basic details about a file such as the resource name, file extension, etc of the file. There is no possibility of getting exceptions here.

File: iamctl/pkg/userStores/import.go (hunk 2 above)
Comment (ImalshaG) [reply iamctl/pkg/userStores/import.go:102]: This is automatically reformatted to this, by the VS code lint extension.

File: iamctl/pkg/utils/commonUtils.go (hunk 3 above)
Comment (ImalshaG) [reply iamctl/pkg/utils/commonUtils.go]: Added a new method `buildRequestUrl` that returns the request url for each request.

File: iamctl/pkg/utils/commonUtils.go (hunk 4 above)
Comment (ImalshaG) [reply iamctl/pkg/utils/commonUtils.go]: Added a new method `buildRequestUrl` that returns the request url for each request.

**Review by chamathns — DISMISSED** [review]

**Review by chamathns — APPROVED** [review]

**Pull request merged**